app/components/QuaternionDisplay/index.js

```diff
@@ -0,0 +1,125 @@
+/**
+*
+* QuaternionDisplay
+*
+*/
+
+import React from 'react';
+import styled from 'styled-components';
+import * as THREE from 'three';
+import 'three-orthographic-trackball-controls';
+
+import { FormattedMessage } from 'react-intl';
+import messages from './messages';
+
+const Wrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+
+  & > canvas {
+    width: 100%!important;
+    height: 100%!important;
+    flex-grow: 1;
+  }
+`;
+
+class QuaternionDisplay extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
```

Comment:
PureComponent uses a shallow compare in its shouldComponentUpdate step, which leads to a reasonable performance boost for components that have constantly changing props/state. In this case that's irrelevant, but there's no reason not to use it here.